src/inmanta/data/model.py

```diff
@@ -702,7 +713,7 @@ class DiscoveredResource(BaseModel):
 
     @field_validator("discovered_resource_id")
     @classmethod
-    def discovered_resource_id_is_resource_id(cls, v: str) -> Optional[Any]:
+    def discovered_resource_id_is_resource_id(cls, v: str) -> str:
```

Comment:
Why not ResourceIdStr?

Reply:
Shouldn't matter too much but it may present a clearer picture. Will do. That or what you proposed on Slack: make the validator an annotator on the `ResourceId` type.